### PR TITLE
[WIP] github: Added pylint action to examine the python code

### PR DESCRIPTION
**Pull request by leemgs**
This commit is to run pyin and send erros into a  PR webpage. It is to improve review effort and code maintenance.

Signed-off-by: Geunsik Lim <leemgs@gmail.com>
Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

### CODE REVIEW

File: .github/workflow/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    name: Executes pylint
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Lint
+        uses: gabriel-milan/action-pylint@v1
+        with:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}} # GitHub token
+          path: "./*.py" # Glob pattern for files to lint
```
Comment (fdegier) [inline .github/workflow/pylint.yml:15]: Not super familiar with GitHub action globbing, but I would expect it to be this pattern based on the documentation https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

```suggestion
          path: "**.py" # Glob pattern for files to lint
```

Comment (leemgs) [reply .github/workflow/pylint.yml]: > Not super familiar with GitHub action globbing, but I would expect it to be this pattern based on the documentation https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

Nice catch. Thanks. Fixed.

**Review by fdegier — APPROVED** [review]

File: .github/workflow/pylint.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    name: Executes pylint
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Lint
+        uses: gabriel-milan/action-pylint@v1
+        with:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}} # GitHub token
+          path: "**.py" # Glob pattern for files to lint
```
Comment (fdegier) [inline .github/workflow/pylint.yml:15]: It looked into it and apparently this is the pattern that pylint can use for global files. 

```suggestion
          path: "**/*.py" # Glob pattern for files to lint
```

**Review by fdegier — COMMENTED** [review]
Just ran pylint locally and it shows a lot of violations, we will need to fix those and add a setup.cfg. I'll follow up with this later

**Review by fdegier — CHANGES_REQUESTED** [review]
Just ran pylint locally and it shows a lot of violations, we will need to fix those and add a setup.cfg. I'll follow up with this later

**Comment by fdegier** [comment]
I just went over the results of pylint and it seems to be very strict, I also ran flake8 and it seems to be more appropriate for the state of this project. I suggest we use Flake8 for now, WDYT?

**Pull request closed without merging**